src/libhirte/bus/utils.h

```diff
@@ -11,27 +11,30 @@ char *bus_path_escape(const char *str);
 typedef struct UnitInfo UnitInfo;
 struct UnitInfo {
         int ref_count;
-
-        const char *node;
-        const char *id;
-        const char *description;
-        const char *load_state;
-        const char *active_state;
-        const char *sub_state;
-        const char *following;
-        const char *unit_path;
+        bool is_from_bus_msg; // Free strings if false
```

Comment:
Is there a case when the `char*`values are set, but `UnitInfo`does not have the ownership of them and, thereore, doesn't need to free them on unref? If not, I think we can remove this variable and always call free (with a check for NULL before).

Reply:
Yeah, when they are assigned from the message. But I will rework this thing